libs/hwes/eventstore.go

```diff
@@ -0,0 +1,92 @@
+package hwes
+
+import (
+	"context"
+	"github.com/EventStore/EventStore-Client-Go/esdb"
+	"github.com/rs/zerolog/log"
+	"hwutil"
+)
+
+var (
+	es *esdb.Client = nil
+)
+
+func SetupEventStoreByEnv() {
+	connectionString := hwutil.MustGetEnv("EVENTSTORE_CS")
+	SetupEventStore(connectionString)
+}
+
+func SetupEventStore(connectionString string) {
+	config, err := esdb.ParseConnectionString(connectionString)
+	if err != nil {
+		log.Fatal().Err(err).Msg("invalid eventstore connection string")
+	}
+
+	client, err := esdb.NewClient(config)
+	_, err = client.ReadAll(context.Background(), esdb.ReadAllOptions{}, 1) // Test connection
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not connect to eventstore")
+	}
+	log.Info().Msg("connected to eventstore")
+	es = client
+}
+
+func AppendEvents(ctx context.Context, id string, events ...Event) error {
+	es := GetES()
+
+	eventsData := make([]esdb.EventData, 0, len(events))
+	for _, event := range events {
+		eventsData = append(eventsData, event.ToEventData())
+	}
+
+	// readOps := esdb.ReadStreamOptions{Direction: esdb.Backwards, From: esdb.End{}}
+	readOps := esdb.ReadStreamOptions{}
+	stream, err := es.ReadStream(ctx, id, readOps, 1)
+	if err != nil && err != esdb.ErrStreamNotFound {
+		return err
+	}
+
+	if err == nil {
+		defer stream.Close()
+	}
+
+	// Stream not exists yet
+	if err == esdb.ErrStreamNotFound {
+		_, err = es.AppendToStream(
+			ctx,
+			id,
+			esdb.AppendToStreamOptions{ExpectedRevision: esdb.NoStream{}},
+			eventsData...,
+		)
+
+		return err
+	}
+
+	lastEvent, err := stream.Recv()
+	if err != nil {
+		return err
+	}
+
+	expectedRevision := esdb.Revision(lastEvent.OriginalEvent().EventNumber)
+
+	_, err = es.AppendToStream(
+		ctx,
+		id,
+		esdb.AppendToStreamOptions{ExpectedRevision: expectedRevision},
```

Comment:
I'm getting a `WrongExpectedStreamRevision` here on `BED_UPDATED_v1` & `BED_DELETED_v1`.

The `BED_UPDATED_v1` is still beeing dispatched and shows up in the db, the `BED_DELETED-v1` not.

Reply:
I've applied some events including finally the `BED_DELETED_v1`, never got this error. Can you check you local setup? If this error still occurs we can hop on a call :)

Reply:
has been fixed by [1ad4583](https://github.com/helpwave/services/pull/582/commits/1ad45834882a80046221afe9cf0d48f2e6bb27d9)